mne/beamformer/_lcmv.py

```diff
@@ -251,8 +283,23 @@ def make_lcmv(info, forward, data_cov, reg=0.05, noise_cov=None, label=None,
                 # finding optimal orientation for NAI and unit-noise-gain
                 # based on [2]_, Eq. 4.47
                 tmp = np.dot(Gk.T, np.dot(Cm_inv_sq, Gk))
-                eig_vals, eig_vecs = linalg.eig(np.dot(linalg.pinv(tmp),
-                                                       np.dot(Wk, Gk)))
+
+                if rank_G == 3:
+                    # use straight inverse with full rank leadfield
+                    eig_vals, eig_vecs = linalg.eig(np.dot(linalg.inv(tmp),
+                                                    np.dot(Wk, Gk)))
+                else:
+                    # use pseudo inverse computation setting smallest component
+                    # to zero if the leadfield is not full rank
+                    eig_vals, eig_vecs = linalg.eig(np.dot(
+                                                    _eig_inv(tmp,
+                                                             range(rank_G)),
+                                                    np.dot(Wk, Gk)))
```

Comment:
can you avoid code duplication by just computing tmp_inv = inv(tmp) the "if else" ?